scripts/db/src/get-user.ts

```diff
@@ -50,48 +64,100 @@ async function getByEmail(email: string, callback: DbScriptCallback) {
       POSTGRES_HOST,
       POSTGRES_DATABASE,
       POSTGRES_PORT,
+      MONGO_URI,
+      MONGO_DB_NAME,
     } = (configuration as unknown) as DbConfiguration
 
-    /** Declare connection info */
-    const connectionInfo: ConnectionConfig = {
-      user: POSTGRES_USERNAME,
-      password: POSTGRES_PASSWORD,
-      host: POSTGRES_HOST,
-      database: POSTGRES_DATABASE,
-      port: POSTGRES_PORT ? parseInt(POSTGRES_PORT) : 5432,
-      ssl: TEMPLATE_DATA.pgShouldSsl,
-    }
+    /**
+     * Logic in this function tries to match:
+     * EAForum/packages/lesswrong/server/vulcan-lib/apollo-server/authentication.tsx
+     */
+    async function getForumUser(): Promise<CallbackUser | null> {
+      // Connect to the Forum DB
+      const mongoClient: MongoClient = new MongoClient(MONGO_URI)
+      await mongoClient.connect()
+
+      // Query the users collection for someone with our email
+      const forumUser = await mongoClient
+        .db(MONGO_DB_NAME)
+        .collection<ForumUser>('users')
+        .findOne({ 'emails.address': email })
+
+      await mongoClient.close()
+
+      if (!forumUser) {
+        return null
+      }
+
+      // Which email did we find?
+      const emailInfo = forumUser.emails.find((e) => e.address === email)
```

Comment:
Not sure what the structure is here, and I assume that a user could have arbitrarily many emails, but if there's a way for Mongo to just return the email field that contains `email`, or if it's guaranteed to only be the first email (and can therefore collapse to `.emails[0]`) etc. then that might be slightly easier.

Reply:
It's not guaranteed to be the first. I could maybe look into trying to query such that it returns only the specified email, but that sounds more complicated than this.